src/sip128.rs

```diff
@@ -378,14 +378,13 @@ impl SipHasher128 {
     }
 
     #[inline]
-    pub fn finish128(mut self) -> [u64; 2] {
```

Comment:
Do we have an idea if this affects performance?

Reply:
I extracted `finish128` method [on Godbolt](https://godbolt.org/z/fd9cWo8K8) and look at the diff with the previous version and there was significant increase in "predicted cycles" and instructions, which was due to the `ptr::write_bytes` call generating a `memcpy` instead of a small instructions because the size wasn't a constant anymore.

I therefore changed that part ([Godbolt](https://godbolt.org/z/v6eEEzWWv)) to copy the "last" and "last + 1" in a array as to have the size a constant as before and now the diff is very small, llvm-mca indicates 15100ins -> 15400ins for 100 iterations.

Reply:
Thanks for investigating!

Since this code is super hot in the compiler and the u64 version of the finish method probably isn't going to be used much, I'd prefer if we didn't regress things at all.

How about extracting a `finish128_inner(nbuf: usize, buf: &mut [MaybeUninit<u64>; BUFFER_WITH_SPILL_CAPACITY], state: State, processed: usize)` function? Then `Hasher::finish` could would only need to copy `state` and `finish128` could keep taking `self` by value (?_

Reply:
I looked at the generated assembly [on Godbolt](https://godbolt.org/z/nW7jvxqnn) and it's very very similar, mainly some registry naming changes, some instructions moving place and one more stack variable.

Compared to the other version (with the last and last+1 copy), it's roughly similar, in terms of instructions per llvm-mca.

One interesting thing, with the inner variant the number of cycles needed per llvm-mca is dropping significantly from 4410 to 3810, maybe because of less memory pressure. That seems like a win.

Pushed the inner variant.